scientific-computing-with-python/learn-tree-traversal-by-building-a-binary-search-tree/binary-search-tree.py

```diff
@@ -0,0 +1,83 @@
+class TreeNode:
+    def __init__(self, key):
+        self.key = key
+        self.left = None
+        self.right = None
+
+class BinarySearchTree:
+    def __init__(self):
+        self.root = None
+
+    def insert(self, key):
+        self.root = self._insert(self.root, key)
+
+    def _insert(self, root, key):
+        if root is None:
+            return TreeNode(key)
+        if key < root.key:
+            root.left = self._insert(root.left, key)
+        else:
+            root.right = self._insert(root.right, key)
+        return root
+
+    def search(self, key):
+        return self._search(self.root, key)
+
+    def _search(self, root, key):
+        if root is None or root.key == key:
+            return root
```

Comment:
Shouldn't search return `True` or `False`, instead of `None` or whatever `key` will be?